### PR TITLE
ci: refactor local cluster tests

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

we try to break local cluster tests into smaller pieces in some previous PRs (https://github.com/solana-labs/solana/pull/22295 and https://github.com/solana-labs/solana/pull/24952)

I think we can only use nextest to group them!

Screenshot 2023-05-23 at 1 49 07 PM
Screenshot 2023-05-23 at 1 49 10 PM

#### Summary of Changes

1. merge `local-cluster/tests/local_cluster_flakey.rs`, `local-cluster/tests/local_cluster_slow_1.rs` and `local-cluster/tests/local_cluster_slow_2.rs` back into `local-cluster/tests/local_cluster.rs`.
2. use nextest to run them partially.

### CODE REVIEW

File: .buildkite/scripts/build-stable.sh [hunk 1]
```diff
@@ -21,6 +21,19 @@ partitions=$(
...
 )
 
+local_cluster_partitions=$(
+  cat <<EOF
+{
+  "name": "local-cluster",
+  "command": ". ci/rust-version.sh; ci/docker-run.sh \$\$rust_stable_docker_image ci/stable/run-local-cluster-partially.sh",
+  "timeout_in_minutes": 30,
+  "agent": "$agent",
+  "parallelism": 5,
```
Comment (yihau) [inline .buildkite/scripts/build-stable.sh:31]: we can scale them up/down easily by updating this number

Comment (yihau) [inline .buildkite/scripts/build-stable.sh:31]: we can scale them up/down easily by updating this number

**Review by CriesofCarrots — COMMENTED** [review]
We used to have issues with local-cluster tests running in parallel, which is why a bunch (all?) of them are marked `[serial]`. Any idea whether we are likely to run into issues with this new approach?

Also, I'm wondering if we want to move these tests forwards to the stable CI block like this. Maybe with the auto-retries flakiness will be less of an issue, but I don't know that local-cluster should block getting to coverage/downstream/etc. Thoughts?

**Comment by yihau** [comment]
> We used to have issues with local-cluster tests running in parallel, which is why a bunch (all?) of them are marked [serial]. 

yeah. It looks like I run them parallel but actually I still run them with `--test-threads=1`. basically I do the same thing like we have `local_cluster_flakey`, `local_cluster_slow-1` and `local_cluster_slow-2` but using the group mechanism from nextest. 

> Any idea whether we are likely to run into issues with this new approach?

yeah. my main purpose to try to make all tests become ~10 mins tests. we can still use the old approach to do it but we will have `local_cluster_slow-3`, `local_cluster_slow-4` .... use this new approach can make scaling easilly.

> Also, I'm wondering if we want to move these tests forwards to the stable CI block like this. Maybe with the auto-retries flakiness will be less of an issue, but I don't know that local-cluster should block getting to coverage/downstream/etc. Thoughts?

I thought local-cluster integration tests as important as stable tests but maybe I'm wrong. we can move them back to the last block in the CI pipeline. the only concern is that I will make those jobs which in the last block more parallel. we may takes lots of agents when our pipeline goes to the last block.

**Review by CriesofCarrots — APPROVED** [review]
Okay, thanks for the details!

> I thought local-cluster integration tests as important as stable tests but maybe I'm wrong. we can move them back to the last block in the CI pipeline. the only concern is that I will make those jobs which in the last block more parallel. we may takes lots of agents when our pipeline goes to the last block.

Fair point. Okay, let's try this as-is, but maybe re-org if these seem to be blocking progress

**Review by t-nelson — APPROVED** [review]
assuming all of the `.rs` changes are just to re-unify the local-cluster tests into a single file, this lgtm

**Pull request merged**